internal/meta/branch.go

```diff
@@ -27,12 +27,24 @@ type Branch struct {
 
 	// The associated pull request information, if any.
 	PullRequest *PullRequest `json:"pullRequest,omitempty"`
+
+	// The commit onto the trunk branch, if any
+	// This will be the merge commit if the PR was merged
+	TrunkCommit *string `json:"trunkCommit,omitempty"`
 }
 
 func (b *Branch) IsStackRoot() bool {
 	return b.Parent.Trunk
 }
 
+func GetStackRoot(repo *git.Repo, name string) Branch {
```

Comment:
I think it would probably be better to just do `GetStackRoot(branches map[string]Branch, name string) Branch`. Every iteration of this would require a `git` command otherwise which builds up to noticeable latency.

Honestly it probably makes sense to change the abstraction we use here at some point so that we can use some kind of "transactionality". I wrote a big comment in `stack_branch.go` a little while ago:
```go
	// TODO:
	// 		It would be really nice to have some kind of atomicity and/or
	//		transactionality with the branch metadatas. Maybe instead of just
	// 		storing each metadata entry as a blob ref, we could store a single
	// 		tree ref that points to each of the branch blobs (since it's
	//	 	possible to update the tree refs atomically w/ Git's CAS).
	// 		It's possible that we run the operations above but then this fails
	//		and we're in a weird state.
```